pkg/platform/types.go

```diff
@@ -16,6 +16,11 @@ type ObjectReference struct {
 type RoutingTarget struct {
 	// ObjectReference provides reference details to the associated object.
 	ObjectReference `json:"ref,omitempty"`
+	// ServiceSelector defines labels used to identify and select the specific service(s)
+	// which belongs to the given the ObjectReference that should be exposed
+	// go expressions are handled in the selector value to set dynamic values from the current ObjectReference;
+	// e.g. {{.metadata.name}}
+	ServiceSelector map[string]string `json:"serviceSelector,omitempty"`
```

Comment:
It is supposed to be `AND` semantic, right?

Worth mentioning significance of lower case, aka it should be correct path?